enhancements/microshift/enabling-user-specfied-featuregates.md

```diff
@@ -0,0 +1,344 @@
+- Verify feature gates are properly formatted in kube-apiserver configuration
+- Test feature gates are correctly applied to kube-controller-manager configuration
+- Validate feature gates are properly set in kube-scheduler configuration
+- Test that feature gates are applied to the correct components based on their scope
+
+### Robot Framework Integration Tests
+
+**Passthrough Verification:**
+- Test that custom feature gates specified in MicroShift configuration appear in component configurations after service restart
+- Verify TechPreviewNoUpgrade preset results in correct feature gates being passed to all components
+- Test CustomNoUpgrade configuration with specific enabled/disabled lists are correctly applied to component configurations
+- Validate that configuration changes only take effect after MicroShift service restart
+
+**Configuration Error Handling:**
+- Test MicroShift behavior with invalid feature gate names (passthrough with component validation)
+- Verify appropriate error reporting when components reject invalid feature gate configurations
+- Test handling of conflicting settings (same feature gate in both enabled and disabled lists)
+
+### Testing Scope Limitations
+
+**Component Behavior Verification:**
+This enhancement does not test whether feature gates actually modify Kubernetes component behavior - that is the responsibility of upstream Kubernetes and OpenShift testing. Testing is limited to verifying the configuration passthrough mechanism works correctly.
+
+**Upgrade Testing:**
+Since upgrades are not supported when custom feature gates are configured, no additional upgrade testing is required for this enhancement. Default upgrade behavior without custom feature gates is already covered by existing MicroShift test suites.
+
+## Graduation Criteria
+
+The feature is planned to be released as GA directly.
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+- Ability to utilize the enhancement end to end
+- End user documentation completed and published
+- Sufficient test coverage including Robot Framework integration tests
+- Available by default
+- End-to-end tests validating configuration passthrough functionality
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+**Default Configuration (no custom feature gates):**
+Upgrades and downgrades proceed normally using standard MicroShift procedures with no additional considerations for feature gate handling.
+
+**Custom Feature Gate Configurations:**
+Upgrades and downgrades are not supported when custom feature gates are configured (TechPreviewNoUpgrade or CustomNoUpgrade). Users must remove all custom feature gate configurations and return to default settings before attempting any version changes.
+
+This limitation aligns with OpenShift's approach where TechPreviewNoUpgrade and CustomNoUpgrade feature sets explicitly prevent cluster upgrades to avoid compatibility issues with experimental features.
```

Comment:
https://github.com/openshift/api/blob/master/config/v1/types_feature.go#L71-L73
It looks like you cannot go back from TP/DP/Custom FGs once enabled.
Undoing FGs on microshift and upgrading would be a deviation